### PR TITLE
Update invoice payment

### PR DESCRIPTION
**Pull request by Tschakki**
- move super.applyAsset() after validation in applyAsset() of Payment transaction.
- refactored code to remove `else` blocks.
- updated `api.js` and removed redundant parameters.
- Updated package.json to use the published lisk-sdk package
- remove unecessary/redundant `client` field from asset

### CODE REVIEW

File: invoice/transactions/payment_transaction.js [hunk 1]
```diff
@@ -15,44 +15,38 @@ class PaymentTransaction extends TransferTransaction {
...
-						transaction.requestedAmount,
-						'Expected amount to be equal or greated than `requestedAmount`',
-					)
-				);
-			}
-		} else {
-			errors.push(
-				new TransactionError(
+		if (!transaction) {
+            return [ new TransactionError(
```
Comment (michielmulders) [inline invoice/transactions/payment_transaction.js]: I prefer to have the errors array as it's easier to understand what this array is doing?

**Review by lsilvs — COMMENTED** [review]
Hi @Tschakki, there are many unrelated changes in this PR. Can you split them to make it easier to review, bitte?
Also, ask @slaweet to review when changes are made in the client.
Cheers!

**Comment by Tschakki** [comment]
@lsilvs ok I put the client-related changes in another branch, there are some problems with them anyway ;)

**Review by michielmulders — APPROVED** [review]

File: invoice/transactions/payment_transaction.js [hunk 2]
```diff
@@ -23,26 +23,31 @@ class PaymentTransaction extends TransferTransaction {
...
+				errors.push(new TransactionError(
+					'RecipientId is not equal to the address that has sent the invoice.',
 					this.id,
-					'.asset.invoiceID',
-					this.asset.data,
-					'Existing invoiceID registered as invoice transaction',
-				)
-			);
+					'.recipientId',
+					transaction.senderId,
```
Comment (lsilvs) [inline invoice/transactions/payment_transaction.js:40]:
```suggestion
					this.recipientId,
```

File: invoice/transactions/payment_transaction.js [hunk 3]
```diff
@@ -23,26 +23,31 @@ class PaymentTransaction extends TransferTransaction {
...
+					'Expected recipientId to be equal to the id of the sender of the invoice.',
```
Comment (lsilvs) [inline invoice/transactions/payment_transaction.js]:
```suggestion
					transaction.senderId,
```

**Review by michielmulders — APPROVED** [review]

**Review by lsilvs — APPROVED** [review]

**Pull request merged**